convert-hf-to-gguf.py

```diff
@@ -215,6 +237,23 @@ def extra_f16_tensors(self, name: str, new_name: str, bid: int | None, n_dims: i
         return False
 
     def write_tensors(self):
+        # same as ggml_compute_fp32_to_bf16 in ggml-impl.h
+        def np_fp32_to_bf16(n: np.ndarray):
+            # force nan to quiet
+            n = np.where((n & 0x7fffffff) > 0x7f800000, (n & 0xffff0000) | (64 << 16), n)
```

Comment:
Everything looks good. This is the only line that bugs me. Not sure about the `(64 << 16)` for logical OR. I'm knee deep in a dataset, so my mental state is not 100% there right now. Could be nothing.

Reply:
It's doing the equivalent of these lines in `ggml-impl.h`:

https://github.com/ggerganov/llama.cpp/blob/befddd0f15de6efb15d7e7f5b527dfb671f4196f/ggml-impl.h#L82-L85

`(64 << 16)` came from adapting this while postponing the right shift until after rounding.
`(n & 0xffff0000)` is to avoid rounding up the NANs by setting the low bits to zero.